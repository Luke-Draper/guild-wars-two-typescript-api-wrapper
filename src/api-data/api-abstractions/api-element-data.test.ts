import MockAPIElementData from "./__mock__/api-element-data";

describe("APIElementData", () => {
	it("instantiates", () => {
		expect(new MockAPIElementData({ id: 1 })).toBeInstanceOf(
			MockAPIElementData
		);
	});
	it("throws error on incorrect information", () => {
		expect(() => {
			return new MockAPIElementData({ id: "fail" });
		}).toThrowErrorMatchingSnapshot();
	});
	it("throws error on no information", () => {
		expect(() => {
			return new MockAPIElementData({});
		}).toThrowErrorMatchingSnapshot();
	});
});
