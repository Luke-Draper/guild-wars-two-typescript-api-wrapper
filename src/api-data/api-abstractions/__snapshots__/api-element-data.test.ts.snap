// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APIElementData throws error on incorrect information 1`] = `"Invalid Data input to APIElementData constructor"`;

exports[`APIElementData throws error on no information 1`] = `"Invalid Data input to APIElementData constructor"`;
